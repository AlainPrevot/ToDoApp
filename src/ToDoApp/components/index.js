export * from './ModeStyle.jsx'
export * from './ToDoAdd.jsx'
export * from './ToDoContentListAdd.jsx'
export * from './ToDoDate.jsx'
export * from './ToDoItem'
export * from './ToDoList.jsx'
export * from './ToDoNav.jsx'
export * from './ToDoOptions.jsx'
export * from './ToDoOptionsButton.jsx'
export * from './ToDoOptionsDropdown.jsx'
