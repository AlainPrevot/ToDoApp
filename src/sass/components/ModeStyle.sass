#mode
    visibility: hidden

.modeDark, .modeLight
    background-position: center
    background-repeat: no-repeat
    background-size: cover
    cursor: pointer
    height: 36px
    position: fixed
    right: 18px
    top: 14px
    width: 36px

.modeLight
    background-image: url('../../Icons/Sol.png')
    display: none

.modeDark
    background-image: url('../../Icons/Luna.png')

#mode:checked ~ .modeLight
    display: block

#mode:checked ~ .modeDark
    display: none
