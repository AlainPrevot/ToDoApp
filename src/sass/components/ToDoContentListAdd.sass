.ToDoContentListAdd
    width: 90%
    height: 70%
    position: relative

.toDoContentListAdd__div
    height: 90%
    overflow-y: scroll

.toDoContentListAdd__div::-webkit-scrollbar
    -webkit-appearance: none

.toDoContentListAdd__div::-webkit-scrollbar:vertical
    width: 4px

.toDoContentListAdd__div::-webkit-scrollbar:horizontal
    height: 2px

.toDoContentListAdd__div::-webkit-scrollbar-thumb
    background-color: var(--Color__secondary)
    border-radius: 20px

.toDoContentListAdd__div::-webkit-scrollbar-track
    border-radius: 10px

.toDoContentListAdd__div::-webkit-scrollbar-button:increment, .toDoContentListAdd__div::-webkit-scrollbar-button
    display: none

@media only screen and (min-width: 560px)

    .toDoContentListAdd__div
        height: 80%
