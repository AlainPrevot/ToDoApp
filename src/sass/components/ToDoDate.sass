.toDoDate
    margin: 50px 0
    text-align: center

    h2
        color: white
        font-size: 2.8rem

    h6
        color: #6E9CB9
        font-size: 1.2rem
        font-weight: 300
        margin: -0.7rem
