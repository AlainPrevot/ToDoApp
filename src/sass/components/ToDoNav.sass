nav
    align-items: center
    display: flex
    justify-content: center
    width: 100%

    a
        color: var(--Color__secondary)
        font-size: 1.4rem
        font-weight: 300
        margin: 6px 10px
        padding: 0 10px
        text-decoration: none
        text-transform: none

        &:hover
            color: var(--font)

.active
    border-bottom: 2px solid var(--Color__primary)
    color: var(--font)
