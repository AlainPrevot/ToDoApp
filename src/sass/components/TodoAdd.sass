
.toDoAdd__input
    background-color: #151D27
    border-radius: 6px
    border: none
    color: white
    height: 46px
    margin-bottom: 20px
    outline: none
    padding-left: 38px
    width: 100%

.toDoAdd__input::placeholder
    color: #6E9CB9
