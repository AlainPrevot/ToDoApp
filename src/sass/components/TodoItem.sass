.toDoItem
    cursor: pointer
    font-size: 1.8rem
    margin-left: 8px

.toDoItem--realizado
    color: var(--Color__primary)

    .toDoItem__span
        text-decoration: line-through

.toDoItem__span
    align-items: center
    display: flex
    font-size: 1rem
    height: 100%
    position: relative
    text-transform: capitalize
    top: -3px
