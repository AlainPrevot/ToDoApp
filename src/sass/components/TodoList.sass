.toDoList
    height: 400px
    overflow-x: hidden
    overflow-y: auto

.toDoList::-webkit-scrollbar
    -webkit-appearance: none

.toDoList::-webkit-scrollbar:vertical
    width: 4px

.toDoList::-webkit-scrollbar:horizontal
    height: 2px

.toDoList::-webkit-scrollbar-thumb
    background-color: var(--Color__secondary)
    border-radius: 20px

.toDoList::-webkit-scrollbar-track
    border-radius: 10px

.toDoList::-webkit-scrollbar-button:increment, .toDoList::-webkit-scrollbar-button
    display: none
