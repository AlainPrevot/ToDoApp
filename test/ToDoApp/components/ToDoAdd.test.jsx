import { fireEvent, render } from "@testing-library/react";
import { ToDoAdd } from "../../../src/ToDoApp/components/ToDoAdd";

/* eslint-disable no-undef */
describe("Pruebas en <ToDoAdd/>", () => {
  test("Debe de hacer math con el Snapshot", () => {
    const onNewTodo = jest.fn();
    const { container } = render(<ToDoAdd onNewTodo={onNewTodo} />);
    expect(container).toMatchSnapshot();
  });

  test("No debe de llamar onNewTodo si el input esta vacio", () => {
    const onNewTodo = jest.fn();
    const { container } = render(<ToDoAdd onNewTodo={onNewTodo} />);
    const form = container.querySelector("form");
    form.submit();
    expect(onNewTodo).toHaveBeenCalledTimes(0);
  });

  test("Debe de llamar a la funcion onNewTodo y enviar el todo", () => {
    const onNewTodo = jest.fn();
    const { container } = render(<ToDoAdd onNewTodo={onNewTodo} />);

    const input = container.querySelector("input");

    const form = container.querySelector("form");

    fireEvent.change(input, { target: { value: "Aprender React" } });

    expect(input.value).toBe("Aprender React");

    fireEvent.submit(form);

    expect(onNewTodo).toHaveBeenCalledTimes(1);

    expect(onNewTodo).toHaveBeenCalledWith({
      id: expect.any(Number),
      done: false,
      description: "Aprender React",
    });
  });

  test("Debe de vaciar el input despues de enviar el todo", () => {
    const onNewTodo = jest.fn();
    const { container } = render(<ToDoAdd onNewTodo={onNewTodo} />);

    const input = container.querySelector("input");

    const form = container.querySelector("form");

    fireEvent.change(input, { target: { value: "Aprender React" } });

    fireEvent.submit(form);

    expect(input.value).toBe("");
  });

  test("Debe de enviar el todo al ejecutar el evento onClick en el button", () => {
    const onNewTodo = jest.fn();
    const { container } = render(<ToDoAdd onNewTodo={onNewTodo} />);

    const input = container.querySelector("input");

    const button = container.querySelector("button");

    fireEvent.change(input, { target: { value: "Aprender React" } });

    fireEvent.click(button);

    expect(onNewTodo).toHaveBeenCalledTimes(1);
  });
});
