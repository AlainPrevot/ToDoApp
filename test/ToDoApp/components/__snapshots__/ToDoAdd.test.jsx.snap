// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ToDoAdd/> Debe de hacer math con el Snapshot 1`] = `
<div>
  <form>
    <input
      autocomplete="off"
      class=" toDoAdd__input"
      name="description"
      placeholder="Add your next task..."
      type="text"
      value=""
    />
    <button
      class="toDoAdd__button"
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;
